File: py-codellama13B-FP-all/taskid-95_check_dict_case-gen16.py
from typing import Dict

def check_dict_case(dict: Dict[str, str]) -> bool:
    """
    Given a dictionary, return True if all keys are strings in lower 
    case or all keys are strings in upper case, else return False.
    The function should return False is the given dictionary is empty.
    Examples:
    >>> check_dict_case({ 'a': 'apple', 'b': 'banana' })
    True
    >>> check_dict_case({ 'a': 'apple', 'A': 'banana', 'B': 'banana' })
    False
    >>> check_dict_case({ 'a': 'apple', 8: 'banana', 'a': 'apple' })
    False
    >>> check_dict_case({ 'Name': 'John', 'Age': '36', 'City': 'Houston' })
    False
    >>> check_dict_case({ 'STATE': 'NC', 'ZIP': '12345' })
    True
    """

    # Check if the dictionary is empty
    if not dict:
        return False
    
    # Check if the dictionary keys are strings
    if not all(isinstance(k, str) for k in dict):
        return False
    
    # Check if all keys are in lower case or all keys are in upper case
    if not all(k.islower() for k in dict) and not all(k.isupper() for k in dict):
        return False
    
    # Otherwise, all keys are in the same case
    return True

File: java-codellama7B-FP-all/taskid-95_check_dict_case-gen16.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // Given a hash map, return true if all keys are strings in lower 
    // case or all keys are strings in upper case, else return false.
    // The function should return false is the given hash map is empty.
    // Examples:
    // >>> checkDictCase((new HashMap<String,String>(Map.of("a", "apple", "b", "banana"))))
    // (true)
    // >>> checkDictCase((new HashMap<String,String>(Map.of("a", "apple", "A", "banana", "B", "banana"))))
    // (false)
    // >>> checkDictCase((new HashMap<String,String>(Map.of("a", "apple", 8l, "banana", "a", "apple"))))
    // (false)
    // >>> checkDictCase((new HashMap<String,String>(Map.of("Name", "John", "Age", "36", "City", "Houston"))))
    // (false)
    // >>> checkDictCase((new HashMap<String,String>(Map.of("STATE", "NC", "ZIP", "12345"))))
    // (true)
    public static boolean checkDictCase(HashMap<String,String> dict) {

        String upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String lower = "abcdefghijklmnopqrstuvwxyz";
        if(dict.isEmpty()) {
            return false;
        }
        boolean keysInLowerCase = true;
        boolean keysInUpperCase = true;
        for(Map.Entry<String,String> entry : dict.entrySet()) {
            if(!entry.getKey().toUpperCase().equals(entry.getKey()) && !entry.getKey().toLowerCase().equals(entry.getKey())) {
                keysInLowerCase = false;
                keysInUpperCase = false;
                break;
            }
            if(entry.getKey().toUpperCase().equals(entry.getKey()) && !entry.getKey().chars().allMatch(c -> upper.indexOf(c) != -1)) {
                keysInLowerCase = false;
            }
            if(entry.getKey().toLowerCase().equals(entry.getKey()) && !entry.getKey().chars().allMatch(c -> lower.indexOf(c) != -1)) {
                keysInUpperCase = false;
            }
        }
        return keysInLowerCase || keysInUpperCase;
    }
    public static void main(String[] args) {
    assert(checkDictCase((new HashMap<String,String>(Map.of("p", "pineapple", "b", "banana")))) == (true));
    assert(checkDictCase((new HashMap<String,String>(Map.of("p", "pineapple", "A", "banana", "B", "banana")))) == (false));
    assert(checkDictCase((new HashMap<String,String>(Map.of("p", "pineapple", "5", "banana", "a", "apple")))) == (false));
    assert(checkDictCase((new HashMap<String,String>(Map.of("Name", "John", "Age", "36", "City", "Houston")))) == (false));
    assert(checkDictCase((new HashMap<String,String>(Map.of("STATE", "NC", "ZIP", "12345")))) == (true));
    assert(checkDictCase((new HashMap<String,String>(Map.of("fruit", "Orange", "taste", "Sweet")))) == (true));
    assert(checkDictCase((new HashMap<String,String>())) == (false));
    }

}
